consul/data_source_consul_config_entry.go

```diff
@@ -0,0 +1,74 @@
+package consul
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceConsulConfigEntry() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceConsulConfigEntryRead,
+
+		Schema: map[string]*schema.Schema{
+			"kind": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The kind of config entry to read.",
+			},
+
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The name of the config entry to read.",
+			},
+
+			"partition": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The partition the config entry is associated with.",
+			},
+
+			"namespace": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The namespace the config entry is associated with.",
+			},
+
+			"config_json": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The configuration of the config entry.",
+			},
+		},
+	}
+}
+
+func dataSourceConsulConfigEntryRead(d *schema.ResourceData, meta interface{}) error {
+	client, qOpts, _ := getClient(d, meta)
+
+	kind := d.Get("kind").(string)
+	name := d.Get("name").(string)
+
+	configEntry, _, err := client.ConfigEntries().Get(kind, name, qOpts)
+	if err != nil {
+		return fmt.Errorf("failed to read config entry %s/%s: %w", kind, name, err)
+	}
+
+	// Config Entries are too complex to write as maps for now so we save their JSON representation
+	data, err := configEntryToMap(configEntry)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s", kind, name))
```

Comment:
Wouldn't id collide with other config entries with the same kind/name in other partitions/namespaces?

Reply:
`id` in Terraform resources or datasources is often unique but does not need to, when objects don't have a specific ID it can happen that they have the same of another. For example Consul has this at https://github.com/hashicorp/terraform-provider-consul/blob/master/consul/resource_consul_keys.go#L188-L191 where all `consul_keys` resources will have the same `id` no matter the keys being set and another occurence of this in the Vault provider is at https://github.com/hashicorp/terraform-provider-vault/blob/main/vault/resource_generic_endpoint.go#L111 where multiple `vault_generic_endpoint` resources can have the same `id` despite being in different namespace.

Here because config entries do not have IDs and the user won't use it it does not matter what placeholder we use.